Doc/library/os.path.rst

```diff
@@ -488,6 +488,41 @@ the :mod:`glob` module.)
       Accepts a :term:`path-like object`.
 
 
+.. function:: splitroot(path)
+
+   Split the pathname *path* into a triad ``(drive, root, tail)`` where:
+
+   1. *drive* is a device name, mount point or the empty string;
+   2. *root* is a sequence of separators following the drive or the empty string; and
+   3. *tail* is anything after the root.
+
+   On POSIX systems, *drive* is always empty. The *root* may be empty (if *path* is
+   relative), a single forward slash (if *path* is absolute), or two forward slashes
+   (implementation-defined per `IEEE Std 1003.1-2017; 4.13 Pathname Resolution
+   <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>`_.)
+   For example::
+
+      >>> splitroot('/etc/hosts')
+      ('', '/', 'etc/hosts')
```

Comment:
```suggestion
      >>> splitroot('home/barney')
      ('', '', 'home/barney')
      >>> splitroot('/home/barney')
      ('', '/', 'home/barney')
      >>> splitroot('//home/barney')
      ('', '//', 'home/barney')      
      >>> splitroot('///home/barney')
      ('', '/', '//home/barney')
```

Reply:
Perhaps getting too long IMHO. @AlexWaygood what do you think? (and below)

Reply:
No other function in this document has a code block with more than two examples, so I agree that four examples here would be too many.

The last example is somewhat surprising (to me at least), so I'd definitely keep that one. Perhaps the second could be skipped.

Reply:
If I had to choose just two Unix examples, I'd opt for the following:

```
      >>> splitroot('/home/barney')
      ('', '/', 'home/barney')
      >>> splitroot('//home/barney')
      ('', '//', 'home/barney')
```

With three examples, I'd add the triple slash example: 

```
      >>> splitroot('///home/barney')
      ('', '/', '//home/barney')
```

I prefer for the examples to be similar in terms of component names, which emphasizes at a glance how the paths are syntactically different in terms of slashes.

Reply:
I'll defer to yours and Barney's superior expertise on paths as to which specific ones are kept, but four is definitely too many for me :) I'd prefer two, but I think I could tolerate three, if necessary.

Reply:
I've opted for 3 so we can include the triple-slash case, which I agree is surprising to newcomers.